Simplenote/SPTextView.m

```diff
@@ -209,4 +209,22 @@ - (NSString *)getLeadingWhiteSpaceForString: (NSString *)string
     return [string substringWithRange:match.range];
 }
 
+- (void)setSelectedRange:(NSRange)selectedRange
+{
+    NSRange patchedSelectionRange = [self fixSelectedRange:selectedRange];
+    [super setSelectedRange:patchedSelectionRange];
+}
+
+// HACK HACK: Only required in macOS Catalina.
+// See Storage.swift `perserveRealEditedRange` for details!
+- (NSRange)fixSelectedRange:(NSRange)selectedRange
```

Comment:
`Storage` internally knows whether it needs to provide the default `selectedRange` or the `overrideSelectionRange` courtesy of the `shouldOverrideSelectionRange`.   It might be nicer if `Storage.selectedRange` could internally check for the flag and return the correct value rather than having to expose the flag to external callers and have the conditional logic below.   The encapsulation would be nicer.  What do you think?

Reply:
That was my first approach... but unfortunately `Storage`'s `super.processEditing()` needs the actual `editedRange` (that is, including the full range that got the right font), otherwise the Emojis go missing.

Storage posts the edited range over to the LayoutManager, which is in charge of updating the TextView's selectedRange. I've **tried** overriding the LM's mechanism, but doing so implies overwriting a method (and **not** calling super, otherwise you get the faulty behavior).

A bit of a cornered situation, I'm afraid. I suspect the true underlying problem are the NSFont warnings we are getting in console (and that's what we've asked Apple about!).

Thanks for the suggestion Eric!!

Reply:
Ouch!  Understood sir.